### PR TITLE
proxy: make latency metrics additive, fix latency calculation for link flow

### PR DESCRIPTION
**Pull request by conradludgate**
## Problem

1. Link flow does not exclude client and cplane time when authenticating
2. Latency metrics are not additive. Prometheus advises that metric labels be additive so that `sum` works properly without specifying any labels.

<https://prometheus.io/docs/practices/naming/#metric-names>
> As a rule of thumb, either the sum() or the avg() over all dimensions of a given metric should be meaningful (though not necessarily useful)

In our current model, summing `excluded=~"client|client_and_cplane|client_cplane_compute|client_cplane_compute_retry"` will equal `4*proxy + 3*retry + 2*compute + 1*cplane` latencies. This goes against the advice above as this is not a meaningful metric

## Summary of changes

1. Add the pause timers for link authenticate
2. Make the latency metrics additive

`{excluded="client_cplane_compute"}` is now equivalent to summing `sum({component=~"proxy|compute_retry"})`

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: proxy/src/auth/backend/link.rs [hunk 1]
```diff
@@ -79,15 +79,19 @@ pub(super) async fn authenticate(
...
     client
         .write_message_noflush(&Be::AuthenticationOk)?
         .write_message_noflush(&Be::CLIENT_ENCODING)?
         .write_message(&Be::NoticeResponse(&greeting))
         .await?;
+    drop(pause);
 
     // Wait for web console response (see `mgmt`).
     info!(parent: &span, "waiting for console's reply...");
+    let pause = ctx.latency_timer.pause(crate::metrics::Waiting::Cplane);
```
Comment (khanova) [inline proxy/src/auth/backend/link.rs:92]: I guess, it's not fully true, that it's Cplane, rather, it's client clicking on the web interface. Probably we could have a single pause.

File: proxy/src/metrics.rs [hunk 2]
```diff
@@ -277,15 +277,16 @@ pub struct ComputeConnectionLatencyGroup {
     protocol: Protocol,
     cold_start_info: ColdStartInfo,
     outcome: ConnectOutcome,
-    excluded: LatencyExclusions,
+    component: LatencyComponents,
 }
 
 #[derive(FixedCardinalityLabel, Copy, Clone)]
-pub enum LatencyExclusions {
+pub enum LatencyComponents {
```
Comment (khanova) [inline proxy/src/metrics.rs:284]: Oh, I thought about it, but, there is no e2e latency anymore (correct me if I am wrong) and I think, it's useful to have it.

Also, with that granularity, how can we measure p90, p99 etc.? I don't think that p90 of Proxy + p90 of Compute is the same as p90 of Proxy and Compute.

Comment (conradludgate) [reply proxy/src/metrics.rs:284]: Let me deploy it to dev-us-east-2 and see if summing over the components gives similar values to what we already observer

File: proxy/src/auth/backend/link.rs (hunk 1 above)
Comment (conradludgate) [reply proxy/src/auth/backend/link.rs:92]: Maybe. I don't think we should worry too much about this though

**Pull request closed without merging**